index.js

```diff
@@ -25,3 +25,21 @@ const port = process.env.port || 3000;
 app.listen(port,()=>console.log(`listening on port ${port}....`));
 // functions
 
+//update function
+app.put('/api/courses/:id',(req,res)=>{
+  const course = courses.find(c=> c.id === parseInt(req.params.id)); 
+  if(!course) res.status(404).send('the course with given ID was not found');
+ 
+  const schema = Joi.object({
+      name: Joi.string().min(3).required()
+  });
+  
+      const result = schema.validate(req.body);
+if(result.error){res.status(400).send(result.error.details[0].message);
+  return;
+}
+course.name = req.body.name;
+res.send(course);

```

Comment:
Tell your colleagues: next task, wrong indentation would result in grades deduction. 😠